includes/ifileio.hpp

```diff
@@ -0,0 +1,15 @@
+#ifndef _IFILEIO_
+#define _IFILEIO_
+
+#include <string>
+#include <vector>
+#include <cstdint>
+
+class IFileIO
+{
+public:
+    virtual int openFile(std::string fileName) = 0;
+    virtual int readFile(uint8_t *buff, int fd) = 0;
```

Comment:
Add some doxygen comment. eg :
int sum(int a, int b);
```
/**
       * A pure function doing a sum of two int variable.
       * @param a an integer argument.
       * @param s an integer arguement
       * @return the sum of \p a and \p s
*/
```
so your readFile would look like this : 
```
int readFile(uint8_t *buff, int fd) = 0;
/**
       * pure Virtual function that reads a file descriptor.
       * @param buff pointer to the output buffer. it will contains what we will read.
       * @param fd the filedescriptor we will read from.
       * @return an int, the number of character read. return zero if there no more character to read.
*/
```

Btw how come you are not specifying how many character you are reading ? 

Maybe don't use a pointer but pass a reference to a vector ? (c++ style)

Reply:
btw do this for most of your classes. usually it's mostly public member function (since they will be used by "others")  :)